src/FlapperUniV2.sol

```diff
@@ -133,9 +135,17 @@ contract FlapperUniV2 {
         emit File(what, data);
     }
 
-    function _getReserves() internal view returns (uint256 reserveDai, uint256 reserveGem) {
+    function _getReserves() internal returns (uint256 reserveDai, uint256 reserveGem) {
         (uint256 _reserveA, uint256 _reserveB,) = pair.getReserves();
         (reserveDai, reserveGem) = daiFirst ? (_reserveA, _reserveB) : (_reserveB, _reserveA);
+
+        uint256 _daiDiff = GemLike(dai).balanceOf(address(pair)) - reserveDai;
```

Comment:
Small nit: maybe save the balances in `_daiBal`, `_gemBal` and use `if(_daiBal > reserveDai || _gemBal > reserveGem) { pair.sync(); reserveDai = _daiBal; reserveGem = _gemDiff; }` to save a a bit on arithmetic ops?

Reply:
Nice. Done.